modules/puzzle/src/main/PuzzleApi.scala

```diff
@@ -85,7 +86,18 @@ private[puzzle] final class PuzzleApi(
             }
     }
 
-    def add(a: Attempt) = attemptColl insert a void
+    def add(a: Attempt) = attemptColl.insert(a) recoverWith lila.db.recoverDuplicateKey { _ =>
+      attemptColl.update($id(a.id),
+        $doc("$set" -> $doc(
+          Attempt.BSONFields.win -> a.win,
+          Attempt.BSONFields.date -> a.date,
+          Attempt.BSONFields.time -> a.time,
+          Attempt.BSONFields.puzzleRating -> a.puzzleRating,
+          Attempt.BSONFields.puzzleRatingDiff -> a.puzzleRatingDiff,
+          Attempt.BSONFields.userRating -> a.userRating,
+          Attempt.BSONFields.userRatingDiff -> a.userRatingDiff
+          )))
+    } void
```

Comment:
One concern with this, is that it is effectively rewriting history.

A frequently requested feature is a training rating graph. Once we change past attempts, it's no longer possible.